content/en/blog/2023/k8s-runtime-observability/index.md

```diff
@@ -0,0 +1,475 @@
+  --volume '[CURRENT_PATH]/config:/etc/kube-tracing@server:*' \
+  --k3s-arg '--kube-apiserver-arg=tracing-config-file=/etc/kube-tracing/apiserver-tracing.yaml@server:*' \
+  --k3s-arg '--kube-apiserver-arg=feature-gates=APIServerTracing=true@server:*' \
+  --k3s-arg '--kubelet-arg=config=/etc/kube-tracing/kubelet-tracing.yaml@server:*'
+```
+
+This command will create a Kubernetes cluster with version `v1.17.1`, and set up
+in three docker containers on your machine. If you run the command
+`kubectl cluster-info` now, you will see this output:
+
+```
+Kubernetes control plane is running at https://0.0.0.0:60503
+CoreDNS is running at https://0.0.0.0:60503/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy
+Metrics-server is running at https://0.0.0.0:60503/api/v1/namespaces/kube-system/services/https:metrics-server:https/proxy
+```
+
+Going back to the logs of the observability environment, you should see some
+spans of internal Kubernetes operations being published in OpenTelemetry
+Collector, like this:
+
+```
+Span #90
+    Trace ID       : 03a7bf9008d54f02bcd4f14aa5438202
+    Parent ID      :
+    ID             : d7a10873192f7066
+    Name           : KubernetesAPI
+    Kind           : SPAN_KIND_SERVER
+    Start time     : 2023-05-18 01:51:44.954563708 +0000 UTC
+    End time       : 2023-05-18 01:51:44.957555323 +0000 UTC
+    Status code    : STATUS_CODE_UNSET
+    Status message :
+Attributes:
+     -> net.transport: STRING(ip_tcp)
+     -> net.peer.ip: STRING(127.0.0.1)
+     -> net.peer.port: INT(54678)
+     -> net.host.ip: STRING(127.0.0.1)
+     -> net.host.port: INT(6443)
+     -> http.target: STRING(/api/v1/namespaces/kube-system/pods/helm-install-traefik-crd-8w4wd)
+     -> http.server_name: STRING(KubernetesAPI)
+     -> http.user_agent: STRING(k3s/v1.27.1+k3s1 (linux/amd64) kubernetes/bc5b42c)
+     -> http.scheme: STRING(https)
+     -> http.host: STRING(127.0.0.1:6443)
+     -> http.flavor: STRING(2)
+     -> http.method: STRING(GET)
+     -> http.wrote_bytes: INT(4724)
+     -> http.status_code: INT(200)
+```
+
+## Testing the Cluster Runtime
+
+With the Observability environment and the Kubernetes cluster set up, you can
+now trigger commands against Kubernetes and see traces of these actions in
+Jaeger.
+
+Open the browser, and navigate to the Jaeger UI located at
+<http://localhost:16686/search>. You’ll see that the `apiserver`, `containerd`,
+and `kubelet` services are publishing traces:
+
+![Jaeger screen with services dropdown open showing apiserver, containerd and kubelet services as options](k8s-services-reported-on-jaeger.png)
+
+Choose `apiserver` and click on **"Find Traces”.** Here you see traces from the
+Kubernetes control plane:
+
+![Jaeger screen showing a list of spans found for apiserver](spans-found-for-apiserver.png)
+
+Let’s run a sample command against Kubernetes with `kubectl`, like running an
+echo:
+
+```bash
+kubectl run -it --rm --restart=Never --image=alpine echo-command -- echo hi
```

Comment:
```suggestion
```console
$ kubectl run -it --rm --restart=Never --image=alpine echo-command -- echo hi
```